pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -1372,28 +1372,36 @@ func validateHugepagesMemoryRequests(field *k8sfield.Path, spec *v1.VirtualMachi
 				Field: field.Child("domain", "hugepages", "size").String(),
 			})
 		} else {
-			vmMemory := spec.Domain.Resources.Requests.Memory().Value()
-			if vmMemory < hugepagesSize.Value() {
+			// Default to spec.domain.resources.requests but use spec.domain.memory.guest if it isn't set as can happen
+			// when this code is called from the VirtualMachine validation webhook before the VirtualMachineInstance mutation
+			// webhook ensures spec.domain.resources.requests is populated.
+			vmMemory := spec.Domain.Resources.Requests.Memory()
+			vmMemoryField := field.Child("domain", "resources", "requests", "memory").String()
+			if vmMemory.IsZero() && spec.Domain.Memory.Guest != nil {
+				vmMemory = spec.Domain.Memory.Guest
+				vmMemoryField = field.Child("domain", "memory", "guest").String()
+			}
```

Comment:
i wonder if similar sorts of issues like this will bite us again in the future.

Maybe we should be running our defaulter on the templated spec (the logic in vmi mutate function) in the vms-admitter.go  before processing the vmi.spec there. would that potentially make sense?

Reply:
@davidvossel Yeah very true, I would however like to land the current limited approach for `v0.59.0` and backport it to at least fix the hugepages corner case while working on a more complete fix for `v0.60.0`, I doubt that fix will be backportable tbh. Would that be acceptable in the short term?

Slightly OT but I think this class of issue also highlights the need to drive more devs and testers to use `VirtualMachines` as their initial starting point in tests etc rather than `VirtualMachineInstances`. I appreciate some of this is a hangover from how things were previously but packages/tools like `libvmi` that encourage the use of `VirtualMachineInstances` and skip `VirtualMachine` validation are starting to hurt us IMHO. If you agree I can take this up on the ML or some other forum (maybe something at the upcoming summit?) to highlight the need for us to exercise `VirtualMachine` validation more.

Reply:
> I doubt that fix will be backportable tbh. Would that be acceptable in the short term?

why wouldn't the systematic fix be backportable?

Reply:
> > I doubt that fix will be backportable tbh. Would that be acceptable in the short term?
> 
> why wouldn't the systematic fix be backportable?

Apologies I misunderstood your suggestion, temporarily applying the defaults in the VM validation webhook would be backportable. I thought you were suggesting we apply them in the VM mutation webhook that while possible would be a pretty big behavioural change and likely result in lots of test fallout. 

I'll start working on this now but I'd still like to land this ahead of `v0.59.0`, I can revert it later as part of the eventual fix and backports if it doesn't make it in this week.